### ISSUE TITLE
Better handle recover of panic

### ISSUE THREAD

**Issue by pzou1974** [keikoproj/addon-manager#18]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG
**What happened**:
sometimes bad addon cause Add-on manager ctl crash.  We should better handle recover of panic
In this case we even can't find failed addon entry and lose the information why addon failed.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Addon Manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- Addon status:
```
$ kubectl describe addon <addon-name>
```
- controller logs:
```
$ kubectl logs <addon-manager-pod>
```

**Comment by pzou1974** [comment]
https://github.com/keikoproj/addon-manager/pull/28

**Comment by kevdowney** [comment]
Closed by #28
